ballista/core/src/client.rs

```diff
@@ -187,3 +204,217 @@ impl RecordBatchStream for FlightDataStream {
         self.schema.clone()
     }
 }
+
+const CONTINUATION_MARKER: [u8; 4] = [0xff; 4];
+
+/// Arrow Stream reader
+pub struct AsyncStreamReader<R: AsyncRead + Unpin + Send> {
```

Comment:
special version of the reader to read object store stream response without memory materialization